networkx/algorithms/tree/operations.py

```diff
@@ -0,0 +1,113 @@
+# operations.py - binary operations on trees
+#
+# Copyright 2015 NetworkX developers.
+#
+# This file is part of NetworkX.
+#
+# NetworkX is distributed under a BSD license; see LICENSE.txt for more
+# information.
+"""Operations on trees."""
+from functools import partial
+from itertools import chain
+
+import networkx as nx
+from networkx.utils import accumulate
+
+__all__ = ['join']
+
+
+def unzip(iterable):
+    """Unzips a zipped iterable.
+
+    For example::
+
+        >>> zipped = list(zip(range(3), range(3)))
+        >>> zipped
+        [(0, 0), (1, 1), (2, 2)]
+        >>> unzipped = list(unzip(zipped))
+        >>> unzipped
+        [(0, 1, 2), (0, 1, 2)]
+
+    """
+    return zip(*iterable)
+
+
+def join(*rooted_trees, label_attribute=None):
```

Comment:
Sadly we still need Python3 so this has to be reworked to process the rooted_trees iterable in the code body.  Else syntax error.